backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+    }
+});
+
+
+// Getting game info
+gameRouter.get('/games/:game_id', async (req,res) => {
+    let result = {};
+    try {
+        const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+        const gameDetails = game.rows[0];
+
+        if (gameDetails.state === 'invited') {
+            return res.status(400).json({ msg: 'Game is in state invited' });
+        };
+
+        let gameUser;
+        let gameOpponent;
+        if (gameDetails.user1 === req.user.id) {
+            gameUser = 'user1'; //string
+            gameOpponent = gameDetails.user2; //id
+        } else {
+            gameUser = 'user2'; //string
+            gameOpponent = gameDetails.user1; //id
+        };
+
+        const usersInformation = await pool.query('select * from users where id = $1', [gameOpponent]);
+
+        // waiting_for_other_player phase
+        if (gameDetails.user1 === req.user.id && gameDetails.state === 'user1_ready' || gameDetails.user2 === req.user.id && gameDetails.state === 'user2_ready') {
+            result = {
+                opponent: usersInformation.rows[0].nickname,
+                phase: 'waiting_for_other_player'
+            }
+        };
+
+        // winner phase
+        if (gameDetails.state === 'user1_won' && gameUser === 'user1'){
+            result = {
+                opponent: usersInformation.rows[0].nickname,
+                phase: 'finished',
+                i_won: true
+            }
+        } else if (gameDetails.state === 'user1_won' && gameUser === 'user2') {
+            result = {
+                opponent: usersInformation.rows[0].nickname,
+                phase: 'finished',
+                i_won: false
+            }
+        } else if (gameDetails.state === 'user2_won' && gameUser === 'user1') {
+            result = {
+                opponent: usersInformation.rows[0].nickname,
+                phase: 'finished',
+                i_won: false
+            }
+        } else if (gameDetails.state === 'user2_won' && gameUser === 'user2') {
+            result = {
+                opponent: usersInformation.rows[0].nickname,
+                phase: 'finished',
+                i_won: true
+            }
+        };
+
+
+        // placing_pieces phase
+        if (gameDetails.state === 'accepted' || (gameDetails.user1 === req.user.id && gameDetails.state === 'user2_ready') || (gameDetails.user2 === req.user.id && gameDetails.state === 'user1_ready')) {
+            const userShips = await pool.query('select * from ships where game_id = $1 and user_id = $2', [req.params.game_id, req.user.id]);
+
+            const gameShips = shipsAmount[gameDetails.dimension];
+            const allGameShips = []; // [5,4,3,3,2]
+            for (key in gameShips) {
+                for (let i = 0; i < gameShips[key]; i++) {
+                    allGameShips.push(key);
+                }
+            };
+
+            if (userShips.rows.length === 0) {
```

Comment:
why do you need this `if`? If there are no ships placed, then the `else` will still be correct (i.e. `placed_ships: []`).